addons/mail/static/src/components/notification_list/tests/notification_list_notification_group_tests.js

```diff
@@ -98,42 +98,33 @@ QUnit.test('notification group basic layout', async function (assert) {
 });
 
 QUnit.test('mark as read', async function (assert) {
-    assert.expect(6);
+    assert.expect(3);
 
     // message that is expected to have a failure
     this.data['mail.message'].records.push({
         id: 11, // random unique id, will be used to link failure to message
         message_type: 'email', // message must be email (goal of the test)
-        model: 'mail.channel', // expected value to link message to channel
+        model: 'sale.order', // expected value to link message to sale.order
         res_id: 31, // id of a random channel
-        res_model_name: "Channel", // random res model name, will be asserted in the test
     });
     // failure that is expected to be used in the test
     this.data['mail.notification'].records.push({
         mail_message_id: 11, // id of the related message
         notification_status: 'exception', // necessary value to have a failure
         notification_type: 'email', // expected failure type for email message
     });
-    const bus = new Bus();
-    bus.on('do-action', null, payload => {
-        assert.step('do_action');
-        assert.strictEqual(
-            payload.action,
-            'mail.mail_resend_cancel_action',
-            "action should be the one to cancel email"
-        );
-        assert.strictEqual(
-            payload.options.additional_context.default_model,
-            'mail.channel',
-            "action should have the group model as default_model"
-        );
-        assert.strictEqual(
-            payload.options.additional_context.unread_counter,
-            1,
-            "action should have the group notification length as unread_counter"
-        );
+    const { createNotificationListComponent } = await this.start({
+        async mockRPC(route, args) {
+            switch (route) {
+                case '/web/dataset/call_kw/sale.order/notify_cancel_by_type':
+                    assert.strictEqual(args.kwargs.notification_type, 'email');
+                    assert.strictEqual(args.model, 'sale.order');
+                    return { result: true }
+                default:
+                    return this._super(...arguments);
+            }
```

Comment:
Using `mockRPC` should be considered deprecated.

What should be done instead is to write the mock_server code for this python method, and then make an actual assert about the notifications being removed from the interface.

There you'll be able to send the bus notification that you need to get a proper update.

Reply:
> Thank you for the update, some follow up comments.

Thank you, I have done the changes accordingly (at least I hope :-))